middleware/tls/tls.go

```diff
@@ -0,0 +1,81 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+)
+
+// NewTLSConfigFromArgs returns a TLS config based upon the passed
+// in list of arguments. Typically these come straight from the
+// Corefile
+func NewTLSConfigFromArgs(args []string) (*tls.Config, error) {
```

Comment:
Think `args ...string` is nicer and conveys the intent slightly better. Also this should have some docs explaining the args....